File: sdks/node/src/domain/resolver/input.ts
interface FileOrigin {
  readonly template: string;
  readonly layer: number;
}

interface ResolvedFile {
  readonly path: string;
  readonly content: string;
  readonly origin: FileOrigin;
}

interface ResolverInput {
  readonly config: Record<string, unknown>;
  readonly files: ResolvedFile[];
}

export type { FileOrigin, ResolvedFile, ResolverInput };

File: sdks/python/cyanprintsdk/domain/resolver/input.py
from dataclasses import dataclass
from typing import Any, Dict


@dataclass
class FileOrigin:
    template: str
    layer: int


@dataclass
class ResolvedFile:
    path: str
    content: str
    origin: FileOrigin


@dataclass
class ResolverInput:
    config: Dict[str, Any]
    files: list[ResolvedFile]
